### ISSUE TITLE
Any option added via elementor/element/section{...} hook is causing preview delay and section jumps

### ISSUE THREAD

**Issue by danyj** [elementor/elementor#7566]
Adding custom options via existing hook is causing uncontrollable jumps, section invisibility and additional delays.
If same set is added to widget everything is smooth. Tested other 3rd party addons that are using these hooks and they all have the same issue. 

https://monosnap.com/file/cK8xPiCVTMrA3G9wf7eAfc0fmNHRYu

```PHP
add_action( 'elementor/element/section/section_background_overlay/after_section_end', array($this,'_add_section_options'), 10, 2 );

public function _add_section_options( $element, $args ) {

	$element->start_controls_section(
		'copt',
		array(
			'label' => __( 'Custom', 'domain' ),
			'tab' => Controls_Manager::TAB_STYLE,
		)
	);

	$element->add_control(
		'c_dimensions',
		array(
			'label'					=>  __( 'Dimensions', 'domain' ),
			'type' 					=> Controls_Manager::SWITCHER,
			'default' 				=> '',
			'label_on' 				=> __( 'Custom', 'domain' ),
			'label_off' 			=> __( 'Default', 'domain' ),
			'return_value' 			=> 'yes',
		)
	);

	$element->add_responsive_control(
		'c_opacity',
		array(
			'label' => __( 'Opacity', 'domain' ),
			'type' => Controls_Manager::SLIDER,
			'size_units' => array('%' ),
			'range' => array(
				'%' => array(
					'min' => 0,
					'max' => 100,
				),
			),
			'devices' => array( 'desktop', 'tablet', 'mobile' ),
			'desktop_default' => array(
				'size' => 100,
				'unit' => '%',
			),
			'tablet_default' => array(
				'size' => 100,
				'unit' => '%',
			),
			'mobile_default' => array(
				'size' => 100,
				'unit' => '%',
			),

		)
	);

	$element->end_controls_section();
}

```

**Comment by bainternet** [comment]
@danyj 

This is happening because you are not instructing the controls what they need to do when a change event happens, for example, the SWITCHER control, if you add `'render_type' => 'none',` to the config array which will tell the control no to do anything on a change event:

```PHP
$element->add_control(
	'c_dimensions',
	[
		'label'        => __( 'Dimensions', 'domain' ),
		'type'         => \Elementor\Controls_Manager::SWITCHER,
		'default'      => '',
		'label_on'     => __( 'Custom', 'domain' ),
		'label_off'    => __( 'Default', 'domain' ),
		'return_value' => 'yes',
		'render_type' => 'none', //<- this line
	]
);
```

or if you tell the control to add a class to the wrapper via `prefix_class` parameter: 

```PHP
$element->add_control(
	'c_dimensions',
	[
		'label'        => __( 'Dimensions', 'domain' ),
		'type'         => \Elementor\Controls_Manager::SWITCHER,
		'default'      => '',
		'label_on'     => __( 'Custom', 'domain' ),
		'label_off'    => __( 'Default', 'domain' ),
		'return_value' => 'yes',
		'prefix_class' => 'custom-dimesions-', //<- this line
	]
);
```

same for the Slider control using `selectors` parameter:
```
$element->add_responsive_control(
    'c_opacity',
    [
        'label' => __( 'Opacity', 'domain' ),
        'type' => \Elementor\Controls_Manager::SLIDER,
        'size_units' => [ '%' ],
        'range' => [
            '%' => [
                'min' => 0,
                'max' => 100,
            ],
        ],
        'devices' => array( 'desktop', 'tablet', 'mobile' ),
        'desktop_default' => [
            'size' => 100,
            'unit' => '%',
        ],
        'tablet_default' => [
            'size' => 100,
            'unit' => '%',
        ],
        'mobile_default' => [
            'size' => 100,
            'unit' => '%',
        ],
        'selectors' => [ // <- this array
            '{{WRAPPER}}' => 'opacity: {{VALUE}}{{UNIT}};',
        ],
    ]
);
```

basically, anything that will tell the control what to do on a change event.

thanks.

**Comment by danyj** [comment]
@bainternet 

The controll needs to do something and if it is on I need 
`'render_type' => 'template'`  to show or hide an html element and yes 'render_type' => 'none'  works 

but than I have no preview. 

Same jumping is if you add the repeater

		$element->add_control(
			'items',
			array(
				'label' => __( 'Background Layers', 'domain' ),
				'type' => \Elementor\Controls_Manager::REPEATER,
				'fields' => $repeater->get_controls(),
				'default' => array(),
				'prevent_empty' => false,
				'render_type' => 'template'
			)
		);

and if I do ` 'render_type' => 'none' ` here no items get displayed in preview
